src/middlewares/imageUploader.js

```diff
@@ -0,0 +1,50 @@
+import Datauri from 'datauri';
+import path from 'path';
+import response from '../utils/response';
+import messages from '../utils/messages';
+import { uploader } from '../config/cloudinaryConfig';
+
+const dUri = new Datauri();
+let ownerId = '';
+
+const extractSingleFile = (rawFile) => {
+  const { originalname, buffer } = rawFile;
+  return dUri.format(path.extname(originalname).toString(), buffer).content;
+};
+
+const extractFiles = (rawFiles) => rawFiles.map(
+  (rawFile) => extractSingleFile(rawFile)
+);
+
+const uploadSingleFile = (file, index) => new Promise((resolve, reject) => {
```

Comment:
why use new promise let use try and catch here.

Reply:
I initially thought I couldn't use `Promise.all()` for `async/await` functions, but I just read up on it and saw that it could be used. Thank you for the feedback, I will alert when I make the changes.

Reply:
Done. I have made the changes